### ISSUE TITLE
Call/Use defined functions/subroutines in integration_tests

### ISSUE THREAD

**Issue by Shaikh-Ubaid** [lfortran/lfortran#1417]
... so that they are not removed by `pass_unused_functions`.

              >I think there exist some integration_tests such that they have functions/subroutines (which are unused) that might fail at the ASR or LLVM level. The integration_tests would pass as such functions/subroutines would be removed by the pass_unused_functions.

I think one such test is https://github.com/lfortran/lfortran/blob/main/integration_tests/operator_overloading_05_module3.f90. I think it might fail if the `pass_unused_functions` is not used. The function here is not being used, so most probably it is removed and there is no llvm being generated.

_Originally posted by @Shaikh-Ubaid in https://github.com/lfortran/lfortran/issues/1412#issuecomment-1462930035_

**Comment by certik** [comment]
Yes, all tests should call the functions and also test the result using an equivalent of "if (a /= 5) error stop".
